software/contrib/consequencer_patterns.py

```diff
@@ -0,0 +1,172 @@
+class pattern:
```

Comment:
With the official repo contrib folder, we are trying to keep scripts contained to a single file so it's easy for all users to simply copy a single script to `main.py`. Can you move the pattern class into your main script, so it fully self contained in a single file? I will update the contrib README to make this more clear in the future.